x-pack/plugins/ingest_manager/dev_docs/schema/saved_objects.mml

```diff
@@ -0,0 +1,73 @@
+classDiagram
+  agent_configs "1" -- "*"  datasources
+  agent_configs "1" -- "*"  enrollment_api_keys
+  agent_configs "1" -- "*"  agents : is used
+  agent_configs "*" -- "*"  outputs
+  agents "1" -- "*"  agent_events
+  agents "1" -- "*"  agent_events
+  package "1" -- "*"  datasources
+
+  class package {
+    installed
+  }
+
+	class agents {
+		status
+    access_api_key_id
+    config_id
+    last_checkin
+    local_metadata
+    user_provided_metadata
+    actions // Encrypted  contains new agent config
+    default_api_key // Encrypted
+
+	}
+
+  class agent_events {
+   type
+    subtype
+    agent_id
+    action_id
+    config_id
+    stream_id
+    timestamp
+    message
+    payload
+    data
+  }
+
+  class agent_configs {
+    datasources // datasource ids
+    name
+    namespace
+    description
+    status
+  }
+
+	class datasources {
+    name
+    namespace
+    config_id
+    enabled
+    package
+    output_id
+    inputs
```

Comment:
I would expand on `inputs` to include the full schema of inputs and streams underneath. I think this is important to know in the context of how we build the resulting agent config.